src/inspect_ai/tool/_tools/_computer/_computer_common.py

```diff
@@ -0,0 +1,127 @@
+import asyncio
+import json
+import logging
+
+from pydantic import BaseModel, Field
+
+from inspect_ai._util.content import ContentText
+from inspect_ai.model import Content, ContentImage
+from inspect_ai.tool._tool import ToolError
+from inspect_ai.tool._tools._computer._mock_logger import MockLogger
+from inspect_ai.util import sandbox
+
+# log = logging.getLogger(__name__)
+log = MockLogger()
+log.setLevel(logging.DEBUG)
+
+
+# TODO: Export ToolResult from inspect_ai.tool
+ToolResult = str | int | float | bool | list[Content]
+
+
+class ShellExecSuccessResult(BaseModel):
+    output: str | None = Field(default=None)
+    base64_image: str | None = Field(default=None)
+
+
+hackIsFirstCommand = True
+
+
+async def _send_cmd(cmdTail: list[str]) -> ToolResult:
+    # TODO: Resolve this issue
```

Comment:
We do actually wait for the health check to pass before starting the sample (for up to 2 minutes): https://github.com/UKGovernmentBEIS/inspect_ai/blob/main/src/inspect_ai/util/_sandbox/docker/compose.py#L27-L39

So if you add a HEALTHCHECK to the Dockerfile that should (🤞 ) take care of this: https://docs.docker.com/reference/dockerfile/#healthcheck